sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/frequency/SketchFrequencies.java

```diff
@@ -0,0 +1,357 @@
+ * {@code PCollection}, or the occurrences of values associated with each key in a
+ * {@code PCollection} of {@code KV}s.
+ *
+ * <p>This class uses the Count-min Sketch structure. The papers and other useful information
+ * about it is available on this website : <a>https://sites.google.com/site/countminsketch/</a>
+ * <br>The implementation comes from Apache Spark :
+ * <a>https://github.com/apache/spark/tree/master/common/sketch</a>
+ */
+public class SketchFrequencies {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SketchFrequencies.class);
+
+  // do not instantiate
+  private SketchFrequencies() {
+  }
+
+  /**
+   * A {@code PTransform} that takes an input {@code PCollection<String>} and returns a
+   * {@code PCollection<CountMinSketch>} whose contents is a Count-min sketch that allows to query
+   * the number of hits for a specific element in the input {@code PCollection}.
+   *
+   * <p>The {@code seed} parameters will be used to randomly generate different hash functions.
+   * Thus, the result can be different for the same stream in different seeds are used.
+   * The {@code seed} parameter will be used to generate a and b for each hash function.
+   * <br>The Count-min sketch size is constant through the process so the memory use is fixed.
+   * However, the dimensions are directly linked to the accuracy.
+   * <br>By default, the relative error is set to 1% with 1% probability that the estimation
+   * breaks this limit.
+   * <br>Also keep in mind that this algorithm works well on highly skewed data but gives poor
+   * results if the elements are evenly distributed.
+   *
+   * <p>See {@link CountMinSketchFn#withAccuracy(double, double)} in order to tune the parameters.
+   * <br>Also see {@link CountMinSketchFn} for more details about the algorithm's principle.
+   *
+   * <p>Example of use:
+   * <pre>{@code
+   * PCollection<String> pc = ...;
+   * PCollection<CountMinSketch> countMinSketch =
+   *     pc.apply(SketchFrequencies.<String>globally(1234));
+   * }</pre>
+   *
+   * <p>Also see {@link CountMinSketchFn} for more details about the algorithm's principle.
+   *
+   * @param seed        the seed used for generating randomly different hash functions
+   */
+  public static Combine.Globally<String, CountMinSketch> globally(int seed) {
+    return Combine.<String, CountMinSketch>globally(CountMinSketchFn
+            .create(seed).withAccuracy(0.001, 0.99));
+  }
+
+  /**
+   * A {@code PTransform} that takes an input {@code PCollection<KV<K, InputT>>} and
+   * returns a {@code PCollection<KV<K, CountMinSketch>>} that contains an output element mapping
+   * each distinct key in the input {@code PCollection} to a structure that allows to query the
+   * count of a specific element associated with that key in the input {@code PCollection}.
+   *
+   * <p>The {@code seed} parameters will be used to randomly generate different hash functions.
```

Comment:
I'd recommend to not duplicate the javadocs so much - it's hard to keep it in sync during review and will be equally hard during future maintenance of this code. All the common stuff you can say in javadoc at class level.